### PR TITLE
Ensure that SplitBrainProtection configs are not overwritten during parsing

### PR DESCRIPTION
**Pull request by pivovarit**
Related: #17874

Depends on: https://github.com/hazelcast/hazelcast/pull/17918

### CODE REVIEW

**Comment by hz-highfive** [comment]
The job `Hazelcast-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/3033/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/3033/Hazelcast-pr-builder-3033.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
-------TEST FAILURE-------
--------------------------
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   XmlOnlyConfigBuilderTest.testConfig2Xml2Config_withAdvancedNetworkConfig:143->testConfig2Xml2Config:226 ? InvalidConfiguration
[ERROR]   XmlOnlyConfigBuilderTest.testConfig2Xml2FullConfig:138->testConfig2Xml2Config:226 ? InvalidConfiguration
[ERROR]   XmlYamlConfigBuilderEqualsTest.testFullConfigAdvancedNetwork:59 ? NullPointer
[ERROR]   XmlYamlConfigBuilderEqualsTest.testFullConfigNormalNetwork:45 ? NullPointer
[ERROR]   YamlConfigBuilderTest.testAllPermissionsCovered:3010 ? NullPointer
[ERROR]   YamlConfigBuilderTest.testConfig_whenProbabilisticSplitBrainProtection_withCustomValues:1882->buildConfig:1692 ? NullPointer
[ERROR]   YamlConfigBuilderTest.testConfig_whenProbabilisticSplitBrainProtection_withDefaultValues:1852->buildConfig:1692 ? NullPointer
[ERROR]   YamlConfigBuilderTest.testConfig_whenRecentlyActiveSplitBrainProtection_withCustomValues:1832->buildConfig:1692 ? NullPointer
[ERROR]   YamlConfigBuilderTest.testConfig_whenRecentlyActiveSplitBrainProtection_withDefaultValues:1811->buildConfig:1692 ? NullPointer
[INFO] 
[ERROR] Tests run: 35383, Failures: 0, Errors: 9, Skipped: 979
[INFO] 

[ERROR] There are test failures.
</details>
</pre>

**Comment by hz-highfive** [comment]
The job `Hazelcast-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/3035/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/3035/Hazelcast-pr-builder-3035.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
-------TEST FAILURE-------
--------------------------
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   XmlYamlConfigBuilderEqualsTest.testFullConfigAdvancedNetwork:59 ? NullPointer
[ERROR]   XmlYamlConfigBuilderEqualsTest.testFullConfigNormalNetwork:45 ? NullPointer
[ERROR]   YamlConfigBuilderTest.testAllPermissionsCovered:3010 ? NullPointer
[ERROR]   YamlConfigBuilderTest.testConfig_whenProbabilisticSplitBrainProtection_withCustomValues:1882->buildConfig:1692 ? NullPointer
[ERROR]   YamlConfigBuilderTest.testConfig_whenProbabilisticSplitBrainProtection_withDefaultValues:1852->buildConfig:1692 ? NullPointer
[ERROR]   YamlConfigBuilderTest.testConfig_whenRecentlyActiveSplitBrainProtection_withCustomValues:1832->buildConfig:1692 ? NullPointer
[ERROR]   YamlConfigBuilderTest.testConfig_whenRecentlyActiveSplitBrainProtection_withDefaultValues:1811->buildConfig:1692 ? NullPointer
[INFO] 
[ERROR] Tests run: 35383, Failures: 0, Errors: 7, Skipped: 979
[INFO] 

[ERROR] There are test failures.
</details>
</pre>

File: hazelcast/src/test/java/com/hazelcast/internal/config/override/ExternalMemberConfigurationOverrideEnvTest.java [hunk 1]
```diff
@@ -142,6 +144,22 @@ public void shouldHandleUserCodeDeploymentConfig() throws Exception {
...
+
+        withEnvironmentVariable("HZ_SPLITBRAINPROTECTION_FOO_ENABLED", "true")
+          .and("HZ_SPLITBRAINPROTECTION_FOO_FUNCTIONCLASSNAME", "com.foo.SomeClass")
+          .execute(() -> new ExternalConfigurationOverride().overwriteMemberConfig(config));
+
+        assertTrue(config.getSplitBrainProtectionConfig("foo").isEnabled());
+        assertSame(SplitBrainProtectionOn.READ, config.getSplitBrainProtectionConfig("foo").getProtectOn());
+        assertEquals("com.foo.SomeClass", config.getSplitBrainProtectionConfig("foo").getFunctionClassName());
+    }
+
```
Comment (blazember) [inline hazelcast/src/test/java/com/hazelcast/internal/config/override/ExternalMemberConfigurationOverrideEnvTest.java:162]: I think we need a test for a non-existent config as well. We rely on getting a non-null config for `config.getSplitBrainProtectionConfig(name)` in `MemberDomConfigProcessor`, but it's not tested.

Comment (pivovarit) [reply hazelcast/src/test/java/com/hazelcast/internal/config/override/ExternalMemberConfigurationOverrideEnvTest.java:162]: There are no non-existent configs. They get instantiated on get (just like our data structures).

Notice that this test actually fetches a _non-existing_ "foo" config at the beginning

Comment (blazember) [reply hazelcast/src/test/java/com/hazelcast/internal/config/override/ExternalMemberConfigurationOverrideEnvTest.java:162]: Ignore this comment, it is tested by the existing config builder tests.

**Review by blazember — APPROVED** [review]

File: hazelcast/src/main/java/com/hazelcast/config/SplitBrainProtectionConfig.java [hunk 2]
```diff
@@ -71,7 +71,9 @@
     private String name;
     private boolean enabled;
     private int minimumClusterSize = 2;
-    private List<SplitBrainProtectionListenerConfig> listenerConfigs = new ArrayList<SplitBrainProtectionListenerConfig>();
+
```
Comment (pivovarit) [inline hazelcast/src/main/java/com/hazelcast/config/SplitBrainProtectionConfig.java:74]: I see that the merge introduced some empty lines here. Will remove them in the next change.

**Pull request merged**